packages/in_app_purchase/lib/src/in_app_purchase_connection.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:async';
+import 'dart:io';
+import 'app_store_connection.dart';
+import 'google_play_connection.dart';
+
+/// Basic generic API for making in app purchases across multiple platforms.
+abstract class InAppPurchaseConnection {
+  /// Returns true if the payment platform is ready and available.
+  Future<bool> isAvailable();
```

Comment:
Suggestion/Question:
I imagine this is something you'll use from Widget build methods to decide e.g whether to show a "buy" button or not, a synchronous API would be easier to use in such cases.
Since it is anyway impossible to guarantee that the value is not stale by the time you get it (as the message channel is async), would it be possible / make sense to maintain an availability status on the dart side that's updated by callbacks and make this synchronous?

[I know nothing about IAP so don't know if this makes sense, really just asking here]

Reply:
I'd want to set up long polling to make sure it didn't go totally stale on the Android song. Maybe I could add it as a configurable option? I'd rather do this in a followup PR. For iOS this is always stable, so I could cache it.

Reply:
I don't think so, but I can investigate for a followup. The only guarantee for `BillingClientStateListener` is that it's talking about setup being complete and successful, specifically. Practically I think it's probably tied to `isReady` but I'm not totally sure. I'd also still need at least one async call here to set it up.

Reply:
ok, lets try to verify that before we first publish (we should be able to return false before we set up the callbacks so this could be synchronous)